File: samples/dianping-php-samples/DemoApiRequest.php
<html>
<head>
<title>Dianping API PHP Samples</title>
<meta content="text/html;charset=utf-8" http-equiv="Content-Type">
</head>
<body>
<?php

//需要PHP 5 以上以及安装curl扩展

//AppKey 信息，请替换
define('APPKEY','XXXXXXXX');

//AppSecret 信息，请替换
define('SECRET','XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX');


//API 请求地址
define('URL', 'http://api.dianping.com/v1/business/find_businesses');

//示例请求参数
$params = array('format'=>'json','city'=>'上海','latitude'=>'31.2204200000392','longitude'=>'121.41163000018','radius'=>'1000','offset_type'=>'1','sort'=>'7','has_deal'=>'1','has_coupon'=>'1','category'=>'美食','region'=>'长宁区','limit'=>'20','keyword'=>'泰国菜');

//按照参数名排序
ksort($params);
//print($params);

//连接待加密的字符串
$codes = APPKEY;

//请求的URL参数
$queryString = '';

while (list($key, $val) = each($params))
{
  $codes .=($key.$val);
  $queryString .=('&'.$key.'='.$val);
}

$codes .=SECRET;
//print($codes);

$sign = strtoupper(sha1($codes));

$url= URL . '?appkey='.APPKEY.'&sign='.$sign.$queryString;

$curl = curl_init();

// 设置你要访问的URL
curl_setopt($curl, CURLOPT_URL, $url);

// 设置cURL 参数，要求结果保存到字符串中还是输出到屏幕上。
curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);

curl_setopt($curl, CURLOPT_ENCODING, 'UTF-8');

// 运行cURL，请求API
$data = json_decode(curl_exec($curl), true);

// 关闭URL请求
curl_close($curl);

print('Your request based on: ');
print('<br/>');
print_r($params);
print('<br/>');
print('Result:');
print('<hr/>');

var_dump($data);
 
?>

</body>
</html>

File: samples/dianping-python-samples/DemoApiRequest.py
#coding=utf-8

import hashlib
import urllib

#请替换appkey和secret
appkey = "XXXXXXXX"
secret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
apiUrl = "http://api.dianping.com/v1/business/find_businesses"

#示例参数
paramSet = []
paramSet.append(("format", "json"))
paramSet.append(("city", "上海"))
paramSet.append(("latitude", "31.21524"))
paramSet.append(("longitude", "121.420033"))
paramSet.append(("category", "美食"))
paramSet.append(("region", "长宁区"))
paramSet.append(("limit", "20"))
paramSet.append(("radius", "2000"))
paramSet.append(("offset_type", "0"))
paramSet.append(("has_coupon", "1"))
paramSet.append(("has_deal", "1"))
paramSet.append(("keyword", "泰国菜"))
paramSet.append(("sort", "7"))

#参数排序与拼接
paramMap = {}
for pair in paramSet:
	paramMap[pair[0]] = pair[1]

codec = appkey
for key in sorted(paramMap.iterkeys()):
	codec += key + paramMap[key]

codec += secret

#签名计算
sign = (hashlib.sha1(codec).hexdigest()).upper()

#拼接访问的URL
url_trail = "appkey=" + appkey + "&sign=" + sign
for pair in paramSet:
	url_trail += "&" + pair[0] + "=" + pair[1]

requestUrl = apiUrl + "?" + url_trail

#模拟请求
response = urllib.urlopen(requestUrl)

print response.read()